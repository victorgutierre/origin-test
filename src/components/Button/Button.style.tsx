import styled from 'styled-components';

export const Btn = styled.button`
  display: block;
  width: 100%;
  max-width: 280px;
  text-align: center;
  cursor: pointer;
  margin: 0 auto;
  padding: 1rem 2rem;
  font-size: 0.9rem;
  font-weight: bold;
  color: #fff;
  border: 0;
  outline: 0;
  border-radius: 30px;
  background-color: #192c8b;

  &:hover {
    background-color: #111f68;
  }
`;

Btn.displayName = 'button';
