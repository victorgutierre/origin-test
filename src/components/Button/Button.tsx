import React from 'react';
import { Btn } from './Button.style';

interface Props {
  label: string;
}

export const Button: React.FC<Props> = ({ label }) => {
  return <Btn data-testid="button">{label}</Btn>;
};
