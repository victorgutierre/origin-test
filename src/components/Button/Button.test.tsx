import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import { Button } from './Button';

describe('<Button />', () => {
  it('Should render the Button component', () => {
    const wrapper = shallow(<Button label="Confirm">Confirm</Button>);
    console.log(wrapper.debug());
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('Label should be "Confirm" same as prop', () => {
    const wrapper = shallow(<Button label={'Confirm'} />);
    const button = wrapper.find('button');
    expect(button.text()).toBe('Confirm');
  });

  it('Label should be a string', () => {
    const wrapper = shallow(<Button label={'Confirm'} />);
    expect(typeof wrapper.props().children).toBe('string');
  });
});
