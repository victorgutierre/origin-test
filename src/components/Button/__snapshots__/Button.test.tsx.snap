// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Should render the Button component 1`] = `
<button>
  Confirm
</button>
`;
